app/services/katello/managed_content_medium_provider.rb

```diff
@@ -0,0 +1,26 @@
+module Katello
+  class ManagedContentMediumProvider < ::MediumProviders::Provider
+    def validate
+      errors = []
+
+      kickstart_repo = entity.try(:content_facet).try(:kickstart_repository) || entity.try(:kickstart_repository)
+
+      errors << N_("Kickstart repository was not set for host '%{host}'") % { :host => entity } if kickstart_repo.nil?
+      errors << N_("Content source was not set for host '%{host}'") % { :host => entity } if entity.content_source.nil?
+      errors
+    end
+
+    def medium_uri(path = "")
+      kickstart_repo = entity.try(:content_facet).try(:kickstart_repository) || entity.try(:kickstart_repository)
```

Comment:
I wonder what happens when the kickstart repository is not set. Where is the mechanism that would fallback to the original Foreman's behavior?

Reply:
It's outside of a specific provider. The providers are chosen based on the result of `#validate` [call](https://github.com/theforeman/foreman/pull/5244/files#diff-21d93ce9e2b033ac75f98c49a726cffcR26).